pulp_ansible/tests/functional/conftest.py

```diff
@@ -99,6 +114,20 @@ def _ansible_repo_factory(**kwargs):
     return _ansible_repo_factory
 
 
+@pytest.fixture
+def ansible_sync_factory(ansible_repo_api_client, ansible_repo_factory):
+    """A factory to perform a sync on an Ansible Repository and return its updated data."""
+
+    def _sync(ansible_repo=None, **kwargs):
+        body = AnsibleRepositorySyncURL(**kwargs)
+        if ansible_repo is None:
+            ansible_repo = ansible_repo_factory()
+        monitor_task(ansible_repo_api_client.sync(ansible_repo.pulp_href, body).task)
+        return ansible_repo_api_client.read(ansible_repo.pulp_href)
+
+    yield _sync
```

Comment:
No need to make it a generator if we don't have cleanup code.
```suggestion
    return _sync
```

Reply:
Would this apply to the other factories that just use `gen_object_with_cleanup` and have no other cleanup code as well?

Reply:
yes, i think so.